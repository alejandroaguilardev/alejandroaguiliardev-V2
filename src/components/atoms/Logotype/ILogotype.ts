export interface ILogotype {
    ligth?:boolean;
}