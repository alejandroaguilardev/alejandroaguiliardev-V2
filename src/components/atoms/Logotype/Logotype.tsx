import Image from "next/image"
import Link from "next/link";
import { Typography, Grid } from "@mui/material"
import { ILogotype } from "./ILogotype";

export const Logotype = ({ ligth = false }: ILogotype) => {
    return (
        <Link href="/">
            <a>
                <Grid sx={{ display: 'flex', alignItems: 'center' }}>
                    <Image
                        src={`${ligth ? '/img/alejandro-aguilar-dev-white.svg' : '/img/alejandro-aguilar-dev.svg'}`}
                        width={50}
                        height={50}
                        alt='Logo Alejandro Aguilar'

                    />
                    <Typography
                        noWrap
                        sx={{
                            mr: 2,
                            fontWeight: 700,
                            letterSpacing: '.2rem',
                            textDecoration: 'none',
                            color: `${ligth ? 'primary.contrastText' : 'text.primary'}`

                        }}
                    >
                        ALEJANDRO AGUILAR
                    </Typography>
                </Grid>
            </a>
        </Link>
    )
}
