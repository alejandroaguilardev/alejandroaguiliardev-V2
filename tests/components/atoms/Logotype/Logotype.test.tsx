import { render } from "@testing-library/react";
import { Logotype } from "../../../../src/components/atoms/Logotype/Logotype";

describe("Logotype", () => {
	test("toMathSnapshot", () => {
		const { container } = render(<Logotype />);
		expect(container).toMatchSnapshot();
	});
});
