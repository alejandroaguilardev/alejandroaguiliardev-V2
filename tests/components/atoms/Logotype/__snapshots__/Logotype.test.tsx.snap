// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype props 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="MuiGrid-root css-2t8riz-MuiGrid-root"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Logo Alejandro Aguilar"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-s53yhl-MuiTypography-root"
      >
        ALEJANDRO AGUILAR
      </p>
    </div>
  </a>
</div>
`;

exports[`Logotype toMathSnapshot 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="MuiGrid-root css-2t8riz-MuiGrid-root"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Logo Alejandro Aguilar"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-s53yhl-MuiTypography-root"
      >
        ALEJANDRO AGUILAR
      </p>
    </div>
  </a>
</div>
`;
